test/test_profiler.py

```diff
@@ -559,7 +559,7 @@ def test_tensorboard_trace_handler(self):
 
         # test case for gzip file format
         with TemporaryDirectoryName() as dname:
-            with profile(
```

Comment:
Don't we also want to keep the old context manager based test?

Reply:
Definitely! We should have both context-manager and non-context-manager tests

Reply:
there are two test in this case. Line 533 use the context manager way and line 562 use the non-context manager ways